ddtrace/internal/datastreams/schemas/schema_sampler.py

```diff
@@ -0,0 +1,25 @@
+import threading
+
+
+class SchemaSampler:
+    SAMPLE_INTERVAL_MILLIS = 30 * 1000
+
+    def __init__(self):
+        self.weight = 0
+        self.last_sample_millis = 0
+        self.lock = threading.Lock()
+
+    def try_sample(self, current_time_millis):
+        if current_time_millis >= self.last_sample_millis + self.SAMPLE_INTERVAL_MILLIS:
```

Comment:
do you know if Python will allow reading this value at the same time as it's written?

Reply:
If the value is being written by an earlier process, then any process trying to read the value during this time that the lock has been acquired will be required to wait until the lock is released by the first process.

Reply:
Discussed this offline, the locking is correct and thread safe.